sources/api/netdog/src/wicked.rs

```diff
@@ -0,0 +1,280 @@
+        let wicked_dhcp6 = config.dhcp6.map(|dhcp6_enabled| WickedDhcp6 {
+            enabled: Some(dhcp6_enabled),
+            _f: (),
+            ..Default::default()
+        });
+        WickedInterface {
+            name,
+            control: WickedControl::default(),
+            ipv4: None,
+            ipv4_dhcp: wicked_dhcp4,
+            ipv6_dhcp: wicked_dhcp6,
+        }
+    }
+
+    /// Serialize the interface's configuration file
+    // Consume `self` to enforce that changes aren't made to the interface type after it has been
+    // written to file
+    pub(crate) fn into_config_file(self) -> Result<()> {
+        let mut cfg_path = Path::new(WICKED_CONFIG_DIR).join(&self.name);
+        cfg_path.set_extension(WICKED_FILE_EXT);
+
+        // TODO: pretty print these files.  `serde_xml_rs` doesn't support pretty printing;
+        // `quick_xml` does, however we require a few features that haven't been released yet to
+        // properly serialize the above data structures:
+        // https://github.com/tafia/quick-xml/issues/340#issuecomment-981093602
+        let xml = serde_xml_rs::to_string(&self).context(error::XmlSerializeSnafu {
+            interface: self.name,
+        })?;
+        fs::write(&cfg_path, xml).context(error::WickedConfigWriteSnafu { path: cfg_path })
+    }
+}
+
+mod error {
+    use snafu::Snafu;
+    use std::io;
+    use std::path::PathBuf;
+
+    #[derive(Debug, Snafu)]
+    #[snafu(visibility(pub(crate)))]
+    pub(crate) enum Error {
+        #[snafu(display("Failed to write network configuration to '{}': {}", path.display(), source))]
+        WickedConfigWrite { path: PathBuf, source: io::Error },
+
+        #[snafu(display("Error serializing config for '{}' to XML: {}", interface, source))]
+        XmlSerialize {
+            interface: String,
+            source: serde_xml_rs::Error,
+        },
+    }
+}
+
+pub(crate) use error::Error;
+type Result<T> = std::result::Result<T, error::Error>;
+
+#[cfg(test)]
+mod tests {
+    use crate::net_config::NetConfig;
+
+    use super::*;
+    use std::path::PathBuf;
+
+    fn test_data() -> PathBuf {
+        PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+            .join("test_data")
+            .join("wicked")
+    }
+
+    #[test]
+    #[cfg(any(bottlerocket_platform = "aws", bottlerocket_platform = "vmware"))]
+    fn default_aws_vmware() {
+        let expected: &str = include_str!("../test_data/wicked/default_aws_vmware.xml");
+
+        let generated = serde_xml_rs::to_string(&WickedInterface::default()).unwrap();
+        assert_eq!(expected.trim(), generated)
+    }
+
+    #[test]
+    #[cfg(bottlerocket_platform = "metal")]
+    fn default() {
+        let expected: &str = include_str!("../test_data/wicked/default.xml");
+
+        let generated = serde_xml_rs::to_string(&WickedInterface::default()).unwrap();
+        assert_eq!(expected.trim(), generated)
+    }
```

Comment:
If you did what I suggested above and conditionally define the constants, then you could conditionally define the test file's path as a constant here in the tests mod. That way there would only be one test for checking the default `WickedInterface`.

Reply:
I see that the `ipv4/6_dhcp` defaults could probably be factored out as well.  That would help a bunch.